custom_components/garmin_connect/services.py

```
@@ -0,0 +1,157 @@
+        muscle_mass = call.data.get("muscle_mass")
+        basal_met = call.data.get("basal_met")
+        active_met = call.data.get("active_met")
+        physique_rating = call.data.get("physique_rating")
+        metabolic_age = call.data.get("metabolic_age")
+        visceral_fat_rating = call.data.get("visceral_fat_rating")
+        bmi = call.data.get("bmi")
+
+        if not await coordinator.async_login():
+            raise HomeAssistantError(
+                "Failed to login to Garmin Connect, unable to update"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)